pandas/tests/io/test_sql.py

```diff
@@ -4358,3 +4358,46 @@ def test_xsqlite_if_exists(sqlite_buildin):
         (5, "E"),
     ]
     drop_table(table_name, sqlite_buildin)
+
+
+@pytest.mark.parametrize("con", all_connectable)
+@pytest.mark.parametrize("dtype_backend", ["pyarrow", "numpy_nullable", lib.no_default])
+def test_bytes_column(con, dtype_backend, request):
+    # GitHub Issue #59242
+    conn = request.getfixturevalue(con)
+    pa = pytest.importorskip("pyarrow")
+
+    dtype = "O"
+    val = b"\x01#Eg\x89\xab\xcd\xef\x01#Eg\x89\xab\xcd\xef"
+
+    if "postgres" in con:
+        val = (
+            b"\x00\x00\x00\x80\x01#Eg\x89\xab\xcd\xef\x01#Eg\x89\xab\xcd\xef"
+            if "adbc" in con
+            else "0000000100100011010001010110011110001001101010"
+            "11110011011110111100000001001000110100010101100"
+            "11110001001101010111100110111101111"
+        )
+        if dtype_backend == "pyarrow":
+            dtype = (
+                pd.ArrowDtype(pa.string())
+                if "adbc" not in con
+                else pd.ArrowDtype(pa.opaque(pa.binary(), "bit", "PostgreSQL"))
```

Comment:
Why is this an opaque type instead of just binary? That seems like a bug somewhere in this implementation or the driver